File: backend/app/schemas/courses.py
import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

CourseLevel = Literal["beginner", "intermediate", "advanced", "expert"]
CourseWriteStatus = Literal["draft", "active"]
CourseStatus = Literal["draft", "active", "archived"]


class CourseWriteRequest(BaseModel):
    title: str = Field(min_length=1, max_length=200)
    description: str = Field(min_length=1)
    category: str = Field(min_length=1, max_length=100)
    level: CourseLevel
    duration_hours: int | None = Field(default=None, ge=0)
    language: str = Field(min_length=1, max_length=50)
    prerequisites: str = Field(default="", max_length=500)
    status: CourseWriteStatus = "draft"


class CourseResponse(BaseModel):
    id: uuid.UUID
    title: str
    description: str
    category: str
    level: CourseLevel
    status: CourseStatus
    duration_hours: int | None
    language: str
    prerequisites: str
    created_at: datetime
    updated_at: datetime

File: frontend/src/lib/courses.ts
export type CourseStatus = 'draft' | 'active' | 'archived';
export type CourseLevel = 'beginner' | 'intermediate' | 'advanced' | 'expert';

export interface Course {
  id: string;
  title: string;
  description: string;
  category: string;
  level: CourseLevel;
  status: CourseStatus;
  durationHours: number | null;
  language: string;
  prerequisites: string;
  updatedAt: string;
}

export const CATEGORY_OPTIONS = [
  'Design',
  'Web Development',
  'Data',
  'Product',
  'Business',
  'Career',
  'Marketing',
];

export const LEVEL_OPTIONS: { value: CourseLevel; label: string }[] = [
  { value: 'beginner', label: 'Beginner' },
  { value: 'intermediate', label: 'Intermediate' },
  { value: 'advanced', label: 'Advanced' },
  { value: 'expert', label: 'Expert' },
];

export const LANGUAGE_OPTIONS = ['English', 'Spanish', 'French', 'German', 'Hindi'];

export const STATUS_OPTIONS: { value: Exclude<CourseStatus, 'archived'>; label: string }[] = [
  { value: 'draft', label: 'Draft' },
  { value: 'active', label: 'Active' },
];

export function formatUpdatedAt(iso: string): string {
  return new Date(iso).toLocaleDateString('en-GB', { day: 'numeric', month: 'short', year: 'numeric' });
}
